js/src/views/Dapps/UrlButton/urlButton.js

```diff
@@ -0,0 +1,94 @@
+// Copyright 2015, 2016 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import React, { Component, PropTypes } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { withRouter } from 'react-router';
+
+import Button from '~/ui/Button';
+import { LinkIcon } from '~/ui/Icons';
+import Input from '~/ui/Form/Input';
+
+import styles from './urlButton.css';
+
+class UrlButton extends Component {
+  static propTypes = {
```

Comment:
router is actually also available on the context, so we don't need the withRouter wrapper. Could just do `static contextTypes = { router: PropTypes.object.isRequired }`, e.g. Same as what we do in `views/Settings/settings.js`